classes/local/opensesame_handler.php

```diff
@@ -510,4 +522,64 @@ private function extract_category_id_from_os_string($stringcategories) {
 
         return $DB->get_field('course_categories', 'id', ['name' => $targetcategory]);
     }
+
+    /**
+     * Generates a name for a course activity.
+     * @param object $opcourse
+     * @return string
+     */
+    public static function generate_activity_name($opcourse) {
+        $pluginconfig = get_config('tool_opensesame');
+        $activityname = $pluginconfig->activity_name;
+        $activityprefix = $pluginconfig->activity_prefix;
+        switch ($activityname) {
+            case 'guid':
+                $name = $opcourse->idopensesame;
+                break;
+            case 'courseid':
+                $name = $opcourse->courseid;
+                break;
+            case 'coursename':
+                $name = $opcourse->title;
+                break;
+            case 'prefix':
+                $name = '';
+                break;
+            default:
+                $name = $opcourse->guid;
+                break;
+        }
+        return !empty($activityprefix) ? $activityprefix . $name : $name;
+    }
+
+    /**
+     * Deletes all disabled course from Moodle.
+     * @return bool If successful.
+     */
+    private function delete_disabled_courses() {
+        global $DB;
+
+        $sql = 'SELECT courseid, status
+                  FROM {tool_opensesame_course}
+                 WHERE active = 0
+                   AND (courseid IS NOT NULL AND courseid <> 0)
+                   AND status <> :status';
+
+        $disabledcourses = $DB->get_records_sql($sql, ['status' => opensesame_course::STATUS_DELETED]);
+
+        foreach ($disabledcourses as $disabledcourse) {
+            !PHPUNIT_TEST ? mtrace('[INFO] Deleting course: ' . $disabledcourse->courseid) : false;
```

Comment:
Could you change this message to "Deleting disabled course"   That will give us more context as to why the course is being deleted.